esx_service/vmdk_ops.py

```diff
@@ -794,78 +871,51 @@ def disk_attach(vmdk_path, vm):
         pvsci = [d for d in controllers
                    if type(d) == vim.ParaVirtualSCSIController and
                       d.key == device.controllerKey]
+        
         return dev_info(device.unitNumber,
-                        get_controller_pci_slot(vm, pvsci,
+                        get_controller_pci_slot(vm, pvsci[0],
                                                 offset_from_bus_number))
+        
 
     # Disk isn't attached, make sure we have a PVSCI and add it if we don't
-    # TODO: add more controllers if we are out of slots. Issue #38
-
     # check if we already have a pvsci one
     pvsci = [d for d in controllers
              if type(d) == vim.ParaVirtualSCSIController]
+    disk_slot = None         
     if len(pvsci) > 0:
-        disk_slot = None  # need to find out
-        controller_key = pvsci[0].key
-        pci_slot_number = get_controller_pci_slot(vm, pvsci,
-                                                  offset_from_bus_number)
-    else:
-        logging.warning(
-            "Warning: PVSCI adapter is missing - trying to add one...")
+        idx, disk_slot = find_available_disk_slot(vm, devices, pvsci, offset_from_bus_number);
+        if (disk_slot is not None):
+            controller_key = pvsci[idx].key
+            pci_slot_number = get_controller_pci_slot(vm, pvsci[idx],
+                                                      offset_from_bus_number)
+            logging.debug("Find an available disk slot, controller_key=%d, slot_id=%d",
+                          controller_key, disk_slot)
+        
+    if (disk_slot is None):
         disk_slot = 0  # starting on a fresh controller
         if len(controllers) >= max_scsi_controllers:
-            msg = "Failed to place PVSCI adapter - out of bus slots"
+            msg = "Failed to place new disk - out of disk slots and out of slots for new controllers"
```

Comment:
I am not sure the error message should mention  "The maximum number of supported volumes of 60 has been reached".  Think about this:

1.  a non PVSCSI controller (LSI SCSI controller) already attached to slot 0

2. attach volumes t1 .. t15,  a new PVSCSI controller will be created at slot 1(controller 1), and volumes t1 .. t15 will be attached to this PVSCSI controller

3. attach volume t16..t30,  another new PVSCSI controller will be created at slot 2(controller 2), and volumes t16 .. t30 will be created to this PVSCSI controller

4. attach volume t31..t45,  another new PVSCSI controller will be created at slot 3(controller 3), and volumes t31 .. t45 will be created to this PVSCSI controller

5. try to attach a new volume t46, which will get this "Failed to place new disk ...." error.

But here, the total number of attached volume is 15*3=45 < 60.

Do we have the assumption that all the 4 SCSI controllers will be PVSCSI controller? 
@msterin @pdhamdhere